src/framework/parsers/texture/texture.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Interface to a texture parser. Implementations of this interface handle the loading
+ * and opening of texture assets.
+ *
+ * @ignore
+ */
+class TextureParser {
+    /* eslint-disable jsdoc/require-returns-check */
+    /**
+     * @function
+     * @name TextureParser#load
+     * @description Load the texture from the remote URL. When loaded (or failed),
+     * use the callback to return an the raw resource data (or error).
+     * @param {object} url - The URL of the resource to load.
+     * @param {string} url.load - The URL to use for loading the resource.
+     * @param {string} url.original - The original URL useful for identifying the resource type.
+     * @param {import('../../../framework/handlers/handler.js').ResourceHandlerCallback} callback - The callback used when
+     * the resource is loaded or an error occurs.
+     * @param {import('../../asset/asset.js').Asset} [asset] - Optional asset that is passed by
+     * ResourceLoader.
+     */
```

Comment:
This can be converted to a 'modern' JSDoc block now (e.g. drop `@function`, `@description` and `@name` tags).